File: www/api/app/Models/ProductItem.php
<?php

namespace App\Models;

use App\Models\Traits\Attributes\ProductItemAttributes;
use App\Models\Traits\EntityPhpDoc;
use App\Models\Traits\Relations\ProductItemRelations;
use Carbon\Carbon;
use Illuminate\Database\Eloquent\Model;

/**
 * @property-read int $id
 * @property int $product_id
 * @property int|null $size_id
 * @property int|null $size_year_id
 * @property int|null $color_id
 * @property int $price_buy
 * @property int $price
 * @property int|null $price_final
 * @property int|null $price_sell
 * @property int $count
 * @property bool $is_sold
 * @property bool $is_for_sale
 * @property bool $is_reserved
 * @property Carbon|null $sold_at
 * @property Carbon|null $created_at
 * @property Carbon|null $updated_at
 */
class ProductItem extends Model
{
    use EntityPhpDoc;
    use ProductItemRelations;
    use ProductItemAttributes;

    /**
     * The attributes that are mass assignable.
     *
     * @var array<int, string>
     */
    protected $fillable = [
        'product_id',
        'size_id',
        'size_year_id',
        'color_id',
        'price_buy',
        'price',
        'price_final',
//        'price_sell',
        'count',
//        'is_sold',
        'is_for_sale',
//        'is_reserved',
    ];

    protected $casts = [
        'sold_at' => 'datetime',
    ];

    public function getMorphClass(): string
    {
        return 'ProductItem';
    }
}

File: www/site-api/models/product_item.go
package models

import (
	"time"
)

// ProductItem представляет конкретный экземпляр товара
type ProductItem struct {
	ID         uint       `json:"id" gorm:"primaryKey"`
	ProductID  uint       `json:"-" gorm:"not null"`
	SizeID     *uint      `json:"-"`
	SizeYearID *uint      `json:"-"`
	ColorID    *uint      `json:"-"`
	PriceBuy   uint       `json:"-" gorm:"not null"`
	Price      uint       `json:"price" gorm:"not null"`
	PriceFinal *uint      `json:"price_final"`
	PriceSell  *uint      `json:"-"`
	Count      uint       `json:"count" gorm:"default:1"`
	IsSold     bool       `json:"is_sold" gorm:"default:false"`
	IsForSale  bool       `json:"is_for_sale" gorm:"default:true"`
	IsReserved bool       `json:"is_reserved" gorm:"default:false"`
	SoldAt     *time.Time `json:"sold_at"`
	CreatedAt  time.Time  `json:"created_at"`
	UpdatedAt  time.Time  `json:"updated_at"`

	// Связи
	Product  Product     `json:"-" gorm:"foreignKey:ProductID;references:ID"`
	Size     *ListOption `json:"size" gorm:"foreignKey:SizeID;references:ID"`
	SizeYear *ListOption `json:"size_year" gorm:"foreignKey:SizeYearID;references:ID"`
	Color    *ListOption `json:"color" gorm:"foreignKey:ColorID;references:ID"`
}

// TableName возвращает имя таблицы для GORM
func (ProductItem) TableName() string {
	return "product_items"
}
